src/codegen/codegen_c_visitor.cpp

```diff
@@ -3736,6 +3746,49 @@ void CodegenCVisitor::print_net_send_buffering() {
 }
 
 
+void CodegenCVisitor::visit_for_netcon(ast::ForNetcon& node) {
+    // For_netcon should take the same arguments as net_receive and apply the operations
+    // in the block to the weights of the netcons. Since all the weights are on the same vector,
+    // weights, we have a mask of operations that we apply iteratively, advancing the offset
+    // to the next netcon.
+    const auto& args = node.get_parameters();
+    RenameVisitor v;
+    auto& statement_block = node.get_statement_block();
+    for (size_t i_arg = 0; i_arg < args.size(); ++i_arg) {
+        // sanitize node_name since we want to substitute names like (*w) as they are
+        auto old_name =
+            std::regex_replace(args[i_arg]->get_node_name(), regex_special_chars, R"(\$&)");
+        auto new_name = "weights[{} + nt->_fornetcon_weight_perm[i]]"_format(i_arg);
+        v.set(old_name, new_name);
+        statement_block->accept(v);
+    }
+
+    const auto index =
+        std::find_if(info.semantics.begin(), info.semantics.end(), [](const IndexSemantics& a) {
+            return a.name == naming::FOR_NETCON_SEMANTIC;
+        })->index;
+    const auto num_int = int_variables_size();
+
+
+    printer->add_line("");  // the first line gets indented in a strange way.
+    if (layout == LayoutType::soa)
+        printer->add_line("//I think it is soa");
+
+
+    std::string offset = (layout == LayoutType::soa) ? "{}*pnodecount + id"_format(index)
+                                                     : "{} + id*{}"_format(index, num_int);
+    printer->add_line("const size_t offset = {};"_format(offset));
```

Comment:
Just a `nitpick` : if you write : 

```
int start = nt->_fornetcon_perm_indices[indexes[offset]];
int end = nt->_fornetcon_perm_indices[indexes[offset]+1];

for (auto i = start; i < end....)
```
it would be bit better for readability?

Reply:
I was worried that "start" and "end" could be already used before in the code...

Reply:
done